### PR TITLE
Few very small refactors

### PR DESCRIPTION
**Pull request by phadej**
Found those while working on parsec parser

### CODE REVIEW

File: Cabal/Distribution/Types/GenericPackageDescription.hs [hunk 1]
```diff
@@ -61,6 +65,9 @@ data Flag = MkFlag
 
 instance Binary Flag
 
+emptyFlag :: FlagName -> Flag
```
Comment (ezyang) [inline Cabal/Distribution/Types/GenericPackageDescription.hs]: Needs doc.

**Comment by ezyang** [comment]
LGTM, but fix the build.

File: Cabal/Distribution/Types/GenericPackageDescription.hs (hunk 1 above)
Comment (phadej) [inline Cabal/Distribution/Types/GenericPackageDescription.hs]: Which would be?

> `Flag`  with default parameters.

?

Comment (ezyang) [inline Cabal/Distribution/Types/GenericPackageDescription.hs]: Well, when I look at the code, I wonder, "Why is it True and then False?" And then I look at `Flag` and I see that the first is `flagDefault` and `flagManual`. At this point I can infer that the default settings for a flag are for it to be (1) turned on by default and (2) not manual (whatever that means...) The comment should explain something like that?

Comment (phadej) [inline Cabal/Distribution/Types/GenericPackageDescription.hs]: would it be clearer to use record syntax for the definition of `emptyFlag`?

Comment (phadej) [inline Cabal/Distribution/Types/GenericPackageDescription.hs]: to avoid boolean blindness, if that's what worries you

Comment (phadej) [inline Cabal/Distribution/Types/GenericPackageDescription.hs]: done

Comment (phadej) [inline Cabal/Distribution/Types/GenericPackageDescription.hs]: ping @ezyang

Comment (ezyang) [inline Cabal/Distribution/Types/GenericPackageDescription.hs]: OK. Could we maybe add Haddock like `-- | A 'Flag' initialized with default parameters (enabled by default, not manual, empty description)`?

Comment (phadej) [inline Cabal/Distribution/Types/GenericPackageDescription.hs]: I'll not repeat the code in the haddock. Comments are IMHO for explaining *why* definition exists or *how" it's meant to be used, not *what* it is (there is "source" for that).
In this case it should be irrelevant, and I unfortunately don't know why default parameters are what they are (but would rather comment that in code, not in haddock).

So it's just ` -- | A 'Flag' initialized with default parameters .`

**Pull request merged**